pkg/limits/frontend/ring_test.go

```diff
@@ -172,11 +142,21 @@ func TestRingStreamUsageGatherer_GetStreamUsage(t *testing.T) {
 			instances := make([]ring.InstanceDesc, len(clients))
 
 			for i := 0; i < len(test.expectedAssignedPartitionsRequest); i++ {
+				expectedStreamUsageReq := (*logproto.GetStreamUsageRequest)(nil)
```

Comment:
What does this do? I'm not sure I follow here

Reply:
Glad that you point that out. The whole loop needs an overhaul here. This setup was written when the two requests types matched one to one.

Reply:
If I delete this, the tests still pass 😕 

```go
clients[i] = &mockIngestLimitsClient{
					expectedAssignedPartitionsRequest: test.expectedAssignedPartitionsRequest[i],
					getAssignedPartitionsResponse:     test.getAssignedPartitionsResponses[i],
					expectedStreamUsageRequest:        test.expectedStreamUsageRequest[i],
					getStreamUsageResponse:            test.getStreamUsageResponses[i],
					t:                                 t,
				}
```